react-calculator/src/App.jsx

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from 'react';
+import './App.css';
+import Digits from './components/digits.jsx';
+import Operations from './components/operations.jsx';
+import { MAX_NUMBER_LENGTH, INDIVISIBLE_NUMBER, RESULT } from './constants.js';
+import store from './utils/store.js';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.resultRef = React.createRef();
+    this.state = {
+      operation: '',
+      firstNumber: store.getLocalStorage('prevResult') ? store.getLocalStorage('prevResult') : '',
+      secondNumber: '',
+      result: store.getLocalStorage('prevResult') ? store.getLocalStorage('prevResult') : '',
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('beforeunload', this.onBeforeUnload);
+  }
+
+  onBeforeUnload = (e) => {
+    e.preventDefault();
+    e.returnValue = '';
+    if (this.state.result === '' || this.state.result === RESULT.RESET) return;
+    store.setLocalStorage('prevResult', this.state.result);
+  };
+
+  handleDigit = (number) => {
+    if (this.state.operation) {
+      const secondNumberResult = this.state.secondNumber + number;
+      this.setState({
+        secondNumber:
+          this.state.secondNumber.length === MAX_NUMBER_LENGTH
+            ? this.state.secondNumber
+            : secondNumberResult,
+      });
+      this.renderCalculatorNumber(secondNumberResult);
+      return;
+    }
+
+    const firstNumberResult = this.state.firstNumber + number;
+    this.setState({
+      firstNumber:
+        this.state.firstNumber.length === MAX_NUMBER_LENGTH
+          ? this.state.firstNumber
+          : firstNumberResult,
+    });
+    this.renderCalculatorNumber(firstNumberResult);
+  };
+
+  setOperations = (selectedOperation) => {
+    this.setState({ operation: selectedOperation });
+  };
```

Comment:
파라미터명을 간단하게 해서 setState() 를 간단하게 해보는건 어떨까요!
```suggestion
  setOperations = (operation) => {
    this.setState({ operation });
  };
```